ext/geos_c_impl/geometry.c

```diff
@@ -420,20 +421,22 @@ method_geometry_disjoint(VALUE self, VALUE rhs)
   self_data = RGEO_GEOMETRY_DATA_PTR(self);
   self_geom = self_data->geom;
   if (self_geom) {
-    rhs_geom = rgeo_convert_to_geos_geometry(self_data->factory, rhs, Qnil);
-    if (rhs_geom) {
+    rhs_geom =
+      rgeo_convert_to_geos_geometry(self_data->factory, rhs, Qnil, &state);
+    if (state) {
+      rb_jump_tag(state);
+    }
 #ifdef RGEO_GEOS_SUPPORTS_PREPARED2
-      prep = rgeo_request_prepared_geometry(self_data);
-      if (prep)
-        val = GEOSPreparedDisjoint(prep, rhs_geom);
-      else
+    prep = rgeo_request_prepared_geometry(self_data);
+    if (prep)
+      val = GEOSPreparedDisjoint(prep, rhs_geom);
+    else
 #endif
-        val = GEOSDisjoint(self_geom, rhs_geom);
-      if (val == 0) {
-        result = Qfalse;
-      } else if (val == 1) {
-        result = Qtrue;
-      }
+      val = GEOSDisjoint(self_geom, rhs_geom);
+    if (val == 0) {
+      result = Qfalse;
+    } else if (val == 1) {
+      result = Qtrue;
```

Comment:
what do you think of

```suggestion

      result = GEOSDisjoint(self_geom, rhs_geom) ? Qtrue : Qfalse
```